src/scripts/directives.js

```diff
@@ -143,10 +143,15 @@
                 },
                 templateUrl: 'templates/type/radio-input.html',
                 link: function (scope) {
-                    scope.localModel = {
-                        value: scope.model
-                    };
-
+                    scope.localModel = '';
+                    var modelWatchDeregister = scope.$watch('model', function (value) {
+                        if (angular.isDefined(value)) {
+                            scope.localModel = {
+                                value: scope.model
+                            };
+                            modelWatchDeregister();
```

Comment:
Why don't you want this watch to stay in place? If the parent attribute changes, you want to update the model inside the directive don't you?